<launch>
  <include file="$(find vicon_bridge)/launch/vicon.launch" />

  <node pkg="asctec_ll" name="hummingbird_1" type="asctec_ll" output="screen">
     <param name="port" value="/dev/ttyUSB0"/>
     <param name="ll_status_rate" value="0.1"/>
     <param name="imu_calc_data_rate" value="0"/>
     <param name="rc_data_rate" value="0"/>
     <param name="ctrl_out_rate" value="5"/>
     <param name="gps_data_advanced_rate" value="0"/>
  </node>

  <node pkg="asctec_ll" name="hummingbird_3" type="asctec_ll" output="screen">
     <param name="port" value="/dev/ttyUSB1"/>
     <param name="ll_status_rate" value="0.1"/>
     <param name="imu_calc_data_rate" value="0"/>
     <param name="rc_data_rate" value="0"/>
     <param name="ctrl_out_rate" value="5"/>
     <param name="gps_data_advanced_rate" value="0"/>
  </node>

  <node name="h1_controller" 
        pkg="pc_asctec_sim" 
        type="pc_asctec_sim_node" 
        output="screen">

     <param name="k_p_xy" value="12.0"/>      <!-- 12.0 -->
     <param name="k_i_xy" value="0.15"/>      <!-- 0.15 -->
     <param name="k_d_xy" value="23.0"/>      <!-- 22.5 -->

     <param name="k_p_z" value="2.35"/>      <!-- 2.55 -->   
     <param name="k_i_z" value="0.18"/>      <!-- 0.18 -->
     <param name="k_d_z" value="3.95"/>       <!-- 3.65 -->

     <param name="k_p_yaw" value="3.0"/>     <!-- 3.0 -->
     <param name="k_i_yaw" value="0.1"/>     <!-- 0.1 -->
     <param name="k_d_yaw" value="1.54"/>    <!-- 0.54 -->

     <param name="quad_name" value="hummingbird_1"/>
     <param name="world_frame" value="/odom"/>
     <param name="quad_frame" value="/vicon/hummingbird_1/hummingbird_1"/>

  </node>

  <node name="h3_controller" 
        pkg="pc_asctec_sim" 
        type="pc_asctec_sim_node" 
        output="screen">

     <param name="k_p_xy" value="12.0"/>      <!-- 12.0 -->
     <param name="k_i_xy" value="0.15"/>      <!-- 0.15 -->
     <param name="k_d_xy" value="23.0"/>      <!-- 22.5 -->

     <param name="k_p_z" value="2.35"/>      <!-- 2.55 -->   
     <param name="k_i_z" value="0.18"/>      <!-- 0.18 -->
     <param name="k_d_z" value="3.95"/>       <!-- 3.65 -->

     <param name="k_p_yaw" value="3.0"/>     <!-- 3.0 -->
     <param name="k_i_yaw" value="0.1"/>     <!-- 0.1 -->
     <param name="k_d_yaw" value="1.54"/>    <!-- 0.54 -->

     <param name="quad_name" value="hummingbird_3"/>
     <param name="world_frame" value="/odom"/>
     <param name="quad_frame" value="/vicon/hummingbird_3/hummingbird_3"/>

  </node>

  <node name="twin_ugv_follower" 
        pkg="pc_asctec_sim" 
        type="twin_ugv_follower" 
        output="screen">
     <param name="world_frame" value="/odom"/>
     <param name="ugv_frame" value="/vicon/jackal2/jackal2"/>
     <param name="h1_name" value="/hummingbird_1"/>
     <param name="h2_name" value="/hummingbird_3"/>
  </node>

</launch>
